caffe2/core/export_caffe2_op_to_c10.h

```diff
@@ -191,48 +194,52 @@ inline FunctionSchema make_function_schema_for_c10(const char* schema_str) {
         ::caffe2::detail::make_function_schema_for_c10(OperatorSchema);       \
     return schema;                                                            \
   }                                                                           \
+  TORCH_LIBRARY_FRAGMENT(_caffe2, m, OperatorName) {                          \
+      m.def(OperatorSchema);                                                  \
+  }                                                                           \
   }                                                                           \
   }
 
 #define C10_EXPORT_CAFFE2_OP_TO_C10_CPU_KERNEL_ONLY(                         \
     OperatorName, OperatorClass)                                             \
   /* Register call_caffe2_op_from_c10 as a kernel with the c10 dispatcher */ \
-  static auto registry_##OperatorName##_##__COUNTER__ =                      \
-      ::c10::RegisterOperators().op(                                         \
-          ::caffe2::_c10_ops::schema_##OperatorName(),                       \
-          ::c10::RegisterOperators::options()                                \
-              .kernel<&::caffe2::detail::call_caffe2_op_from_c10<            \
-                  ::caffe2::_c10_ops::schema_##OperatorName,                 \
-                  OperatorClass>>(::c10::DispatchKey::CPU));
-
-#define C10_EXPORT_CAFFE2_OP_TO_C10_CPU(                                \
-    OperatorName, OperatorSchema, OperatorClass)                        \
-  C10_EXPORT_CAFFE2_OP_TO_C10_SCHEMA_ONLY(OperatorName, OperatorSchema) \
+    TORCH_LIBRARY_IMPL_FRAGMENT(_caffe2, CPU, m, OperatorName) {             \
+        m.impl("_caffe2::" #OperatorName,                                    \
```

Comment:
Just want to call out that I'm hardcoding the namespace (_caffe2::) directly in the macro, since passing it in would require modifying all of the call-sites. I think this is fine, but there are a handful of caffe2 ops that actually don't obey this convention- they register an op twice, once under `_caffe2` and a second time under `__caffe2`. A grep/fbgs search told me that the ops in the second namespace aren't actually used anywhere, so for now I'm deleting them (see `caffe2/operators/bbox_transform_op.cc`)

Reply:
A safer way to stage this change would be to do a small diff just removing these `__caffe2` registrations, with a heads up to mobile team (cc'ing @linbinyu who added one of these, and @iseeyuan reviewer), wait a little bit, and then land the rest of the changes.